pypeit/spectrographs/spectrograph.py

```diff
@@ -676,55 +666,213 @@ def get_detector_par(self, det, hdu=None):
+              set of detector and spatial pixel coordinate combinations needed
+              to reduce a single slit; see ``slitspatnum`` in the
+              :ref:`pypeitpar`.  The string is parsed into the list of relevant
+              detectors.
+
+            - If ``subset`` is a list, integer, or tuple, it is parsed into a
+              set of single detector or detector mosaics to reduce.
+
+        Args:
+            subset (:obj:`int`, :obj:`tuple`, :obj:`list`, :obj:`str`, optional):
+                An object used select a subset of detectors to reduce.  See
+                description above.  Note detectors are 1-indexed.
+
+        Returns:
+            :obj:`list`: List of detectors or detector mosaics to be reduced.
+
+        Raises:
+            PypeItError: Raised if any of the detectors or detector mosaics
+            specified by ``subset`` are invalid.
+        """
+        if subset is None:
+            return np.arange(1, self.ndet+1).tolist()
+
+        if isinstance(subset, str):
+            _subset = parse.parse_slitspatnum(subset)[0].tolist() 
+        elif isinstance(subset, (int, tuple)):
+            _subset = [subset]
+        else:
+            _subset = subset
+
+        allowed = np.arange(1, self.ndet+1).tolist() + self.allowed_mosaics
+        if any([s not in allowed for s in _subset]):
+            msgs.error('Selected detectors or detector mosaics contain invalid values.')
+
+        return _subset
+
+    @property
+    def default_mosaic(self):
+        """
+        Return the default detector mosaic.
+
+        For instruments with no allowed detector mosaics, this *must* be
+        returned as None.
+        """
+        return None
+
+    def validate_det(self, det):
+        """
+        Validate the detector(s) provided.
+
+        Args:
+            det (:obj:`int`, :obj:`tuple`):
+                Selection of 1-indexed detector(s).  An image mosaic is selected
+                using a :obj:`tuple` with the detectors in the mosaic, which
+                must be one of the allowed mosaics returned by
+                :func:`allowed_mosaics`.
+
+        Returns:
+            :obj:`tuple`: Returns the number of detectors and a tuple with the
+            validated detectors.  If ``det`` is provided as a single integer,
+            the latter is just a one-element tuple.  Otherwise, it is identical
+            to the input ``det`` tuple, assuming the validation was successful.
+        """
+        if isinstance(det, tuple):
+            if det not in self.allowed_mosaics:
+                msgs.error(f'Selected detectors {det} are not an allowed mosaic for {self.name}.')
+            return len(det), det
+        if not isinstance(det, (int, np.integer)):
+            msgs.error(f'Provided det must have type tuple or integer, not {type(det)}.')
+        return 1, (det,)
+
     def get_rawimage(self, raw_file, det):
```

Comment:
It seems to me like this and associated raw image reading methods should be functions in core and only wrapped here in spectrograph. I know that is not the subject of this PR, but maybe we can make an issue on that.

Reply:
That's a pretty significant refactor.  I'll leave it to you to raise an issue.